dartagnan/src/main/java/com/dat3m/dartagnan/verification/solving/RefinementSolver.java

```diff
@@ -180,10 +180,15 @@ private void runInternal(SolverContext ctx, ProverEnvironment prover, Verificati
 
         // ------------------------ Preprocessing / Analysis ------------------------
 
-        removeFlaggedAxioms(memoryModel);
+        // TODO: This is a reasonable transformation for all solvers, however,
+        //  our current processing pipelines (WmmProcessor/ProgramProcessor) are unaware of the property
+        //  so we cannot perform property-aware transformation in those pipelines right now.
+        removeFlaggedAxiomsIfNotNeeded(task);
```

Comment:
It means that this transformation of the memory model (deleting all flagged axioms) could be used for all solvers, including the eager one. So a reasonable choice would be to have a `WmmProcessor` that does this transformation generally, however, this `WmmProcessor` would need to be aware of the specification we check for (if `cat_spec` don't touch the memory model, otherwise delete flagged axioms).
And this "specification-awareness" is something that we do not have right now.
It would be easy if the `WmmProcessor` took as input not the `Wmm` but the whole `VerificationTask`.

Reply:
Ok, then please use "both our methods, i.e. eager and lazy" ... Solver is a term quote overloaded in our code base